Add warning about scoped Slot implementation

LinusBorg opened the pull request.

Commit:
docs(mocha-webpack): replace mocha-webpack by mochapack fork

LinusBorg committed:
Add warning about scoped Slot implementation

LinusBorg committed:
Add warning about scoped Slot implementation

schumilin committed:
docs: update Karma docs link (#1208)

Jinjiang committed:
docs(zh): update translation (#1194)

wiese commented on docs/api/options.md:
````diff
@@ -124,6 +124,28 @@ shallowMount(Component, {
 })
 ```
+::: warning Root Element required
+Due to the internal implementatioj of this feature, the slot content has to return a root element, even though a scoped slot is allowed to return an array of elements.
````
```suggestion
Due to the internal implementation of this feature, the slot content has to return a root element, even though a scoped slot is allowed to return an array of elements.
```

wiese committed:
Update docs/api/options.md

correct typo

Co-Authored-By: LinusBorg <t.luenborg@googlemail.com>

wiese committed:
Update docs/api/options.md

correct typo

Co-Authored-By: LinusBorg <t.luenborg@googlemail.com>

eddyerburgh commented:
Thanks @LinusBorg . Can you run prettier to fix the formatting issue?

LinusBorg commented:
@eddyerburgh 

* when i run prettier on the whole repo (`yarn format`) it changes >200 files.
* When I run it on just the docs file (yarn prettier --no-semi --single-quote --write "./docs/api/options"`) it also changes around 20 other things, mostly turning `- ` into `* ` for lists.

No idea why?

LinusBorg committed:
adjust formatting

LinusBorg committed:
adjust formatting

Commit:
docs: update Russian translation (#1215)

* docs: update russian translation

* style: format docs pages

* chore(docs): typo

Commit:
refactor(types): use Component definition from Vue core (#1213)

dnutiu committed:
docs: add example for trigger with modifier keys (#1203)

Commit:
docs(wrapper): сlarify attachedToDocument meaning in docs (#1214)

markbrouch committed:
fix: polyfill Element.matches for IE < 11 (#1230)

fixes #1223

garyo committed:
fix(types): allow `false` as a component stub value (#1231)

addisonssense committed:
feat: add option to pretty print html components (#1229)

BREAKING CHANGE: html output will now be formatted

winniehell committed:
chore: Rename selector to selectorString in ErrorWrapper

winniehell committed:
chore: Build selector string in ErrorWrapper

winniehell committed:
feat: Add selector property to BaseWrapper interface

winniehell committed:
feat: Store selector in Wrapper.find() / .findAll()

winniehell committed:
chore: Remove selectorString from ErrorWrapper

7inspire committed:
docs: update common-tips.md (#1249)

winniehell committed:
chore: use config file for Prettier (#1247)

winniehell commented:
@LinusBorg Just a wild guess but maybe https://github.com/vuejs/vue-test-utils/pull/1247 helps here. :thinking:

stevegrunwell committed:
fix: respect "hidden" attributes in isVisible() (#1257)

Elements that are hidden via the HTML `hidden` attribute were being reported as being
visible by the `isVisible()` method, even though they were not actually visible in the DOM.

Jinjiang committed:
docs(zh): updated (#1263)

atilacamurca committed:
docs: update common-tips.md (#1290)

LionKenedi95 committed:
docs: correct `$nexTick` -> `$nextTick` (#1282)

piktur committed:
fix: add $set scoped slot helper (#1287)

fixes #1253

livesega committed:
Fixed typo (#1296)

tomasbjerre committed:
docs: fix Lerna link in CONTRIBUTING.md (#1299)

lmillucci committed:
chore: bump lodash to ^4.17.15 to fix vulnerability (#1297)

Fixes #1275

alexbonhomme committed:
Merge branch 'dev' into dev

earnubs committed:
Allow selectors in template option when testing

xhwgood committed:
docs(cn): fix error in testing-async-component.md (#1343)

vvanpo committed:
Update dependencies and compatibility tests (#1341)

* Update lockfile

`yarn install` was failing with newer node versions (>=10) because the
version of fsevents was too old (no precompiled binaries), and node-gyp
would fail to build it.

`yarn upgrade` mitigates this issue, but a change in `prettier` 1.18
(https://prettier.io/blog/2019/06/06/1.18.0.html) required `yarn format`
to also be run.

A missing dependency (`rollup-plugin-json`) was added to
`@vue/test-utils`, as it is imported in its build script.

* Update compatibility test versions

Update minor versions of running `yarn test:compat` to the latest patch
revisions.

Add optional argument to `yarn test:compat`, to allow running tests
against arbitrary `vue` versions.

Run compatibility tests in reverse order, to ensure the most relevant
test failures are displayed.

Run `yarn add` non-interactively, to force errors if `yarn test:compat`
is passed an invalid value.

Run `yarn add` without clobbering the lockfile.

* Update CircleCI config for compatibility tests

Having each compatibility test version specified in both the CI config
and the package script leads to inconsistencies over time. The CI config
can just use the package script.

* Hoist dev dependencies explicitly

Common build dependencies across sub-packages can get out-of-sync over
time. Currently, `rollup` and plugins is used across both `test-utils`
and `server-test-utils`, but different versions are specified. This
requires duplicate installations, and prevents hoisting.

Since `devDependencies` are not present in published packages, and since
the monorepo is used exclusively for building the subpackages, it makes
sense to define all build dependencies in the root package.

Move all `devDependencies` in the root package to `dependencies`, to
avoid unmet peer dependency errors in `yarn`:
https://github.com/yarnpkg/yarn/issues/5810

Update each of these dependencies to their latest version, in particular
`rollup`, which has since released stable versions.

See https://github.com/lerna/lerna/blob/master/doc/hoist.md#lerna-hoisting

* Remove build step from compatibility tests

Each compatibility test installs a legacy version of `vue`, and builds
each package against this version before running the test suites. This
is assumedly in error, as the only build that gets published is the one
built against the dependencies in the lockfile.

To properly test compatibility for consumers using older versions of
`vue` to satisfy each package's peer dependency, we need to test against
the build that would actually get published. The fix is to run `yarn
build:test` only once, before any compatibility tests are run.

* Relax rollup dependency constraints

The upgrade from `rollup` beta -> stable went without breakages, so
clearly we don't need to overly constrain updated dependencies using the
`^` semver specifier.

* Revert locked dependencies

The updates from 1d6950af have to be partially reverted to allow for the
last compatiblity test suite to pass: `vue@2.0.8`. There is some unclear
combination of dependencies that causes 80 unit test failures for only
the 2.0 version of vue.

vvanpo committed:
Use `mount()` consistently in Wrapper documentation (#1340)

The `setSelected()` documentation examples imported `shallowMount`, but
called `mount()`.

karatekaneen committed:
Added async/await example (#1338)

Added example of using async/await on functions that are using Promises under the hood.

nonsocode committed:
Fixed syntax mistake (`shallowMount` => `mount`) (#1336)

`shallowMount` doesn't render children components so the example would not have worked. I changed it to `mount`

Commit:
Add link to trigger events. (#1324)

To make it easier to see which DOM events can be triggered.

dependabot[bot] committed:
chore(deps): bump sshpk from 1.13.1 to 1.16.1 (#1347)

Bumps [sshpk](https://github.com/joyent/node-sshpk) from 1.13.1 to 1.16.1.
- [Release notes](https://github.com/joyent/node-sshpk/releases)
- [Commits](https://github.com/joyent/node-sshpk/compare/v1.13.1...v1.16.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump fstream from 1.0.11 to 1.0.12 (#1346)

Bumps [fstream](https://github.com/npm/fstream) from 1.0.11 to 1.0.12.
- [Release notes](https://github.com/npm/fstream/releases)
- [Commits](https://github.com/npm/fstream/compare/v1.0.11...v1.0.12)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump mixin-deep from 1.3.1 to 1.3.2 (#1345)

Bumps [mixin-deep](https://github.com/jonschlinkert/mixin-deep) from 1.3.1 to 1.3.2.
- [Release notes](https://github.com/jonschlinkert/mixin-deep/releases)
- [Commits](https://github.com/jonschlinkert/mixin-deep/compare/1.3.1...1.3.2)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump macaddress from 0.2.8 to 0.2.9 (#1344)

Bumps [macaddress](https://github.com/scravy/node-macaddress) from 0.2.8 to 0.2.9.
- [Release notes](https://github.com/scravy/node-macaddress/releases)
- [Commits](https://github.com/scravy/node-macaddress/compare/0.2.8...0.2.9)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump eslint from 4.18.1 to 4.18.2 (#1350)

Bumps [eslint](https://github.com/eslint/eslint) from 4.18.1 to 4.18.2.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v4.18.1...v4.18.2)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump stringstream from 0.0.5 to 0.0.6 (#1349)

Bumps [stringstream](https://github.com/mhart/StringStream) from 0.0.5 to 0.0.6.
- [Release notes](https://github.com/mhart/StringStream/releases)
- [Commits](https://github.com/mhart/StringStream/compare/v0.0.5...v0.0.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump lodash.template from 4.4.0 to 4.5.0 (#1348)

Bumps [lodash.template](https://github.com/lodash/lodash) from 4.4.0 to 4.5.0.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.4.0...4.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

WhoisUnknown committed:
Update props.md. Text correction (#1310)

AleksandrSl committed:
Fix typos in russian translation of options.md (#1334)

Meowu committed:
fix: typo. (#1312)

cjpearson committed:
add favicon (#1211)

eddyerburgh committed:
Add guide on next tick (#1156)

* docs: update getting started

* docs: update common tips

* docs: update testing-async-components

* docs: remove details on sync updating

* Update testin-async-components

* docs: update common-tips section

* docs: reword getting-started

* docs: reword

* docs: update getting-started

* docs: reword

* docs: reword

* docs: update example

eddyerburgh committed:
docs: remove sync mode from docs (#1142)

* docs: remove sync mode

* docs: remove TransitionGroupStub and TransitionStub pages

* refactor: remove dist dir

* docs: tweak

Commit:
Support composition api variable for  provider

Commit:
Support composition api variable for  provider

Commit:
Add files for flow support in vscode and updated flow to 0.111.x

Commit:
Add files for flow support in vscode and updated flow to 0.111.x

Commit:
Add test for injection on composition api component.

Commit:
Add test for injection on composition api component.

Commit:
Fix prettier on vscode workspace settings.

Commit:
Fix prettier on vscode workspace settings.

Commit:
More prettier changes

Commit:
More prettier changes

Commit:
Removed unused imports from injection composition vue file

Commit:
Removed unused imports from injection composition vue file

Commit:
Even more  prettier.

Commit:
Even more  prettier.

eddyerburgh committed:
refactor: format

eddyerburgh committed:
chore(release): publish v1.0.0-beta.30

eddyerburgh committed:
refactor: format

eddyerburgh committed:
chore(release): add v1.0.0-beta.30 dist files

hiendv committed:
fix: whitelist tags to be ignored with `disabled` attribute

lmiller1990 committed:
(docs): Add guide for some useful libraries for testing

lmiller1990 committed:
(docs): add a guide on testing without a build step in node.js

lmiller1990 committed:
(docs): fix linting in docs

lmiller1990 committed:
(docs): run prettier

lmiller1990 committed:
(docs): run formatter

lmiller1990 committed:
(docs): change how we refer to vue-test-utils

lmiller1990 committed:
(docs): refer to Vue Test Utils correctly

earnubs committed:
Make conditional spec easier to read

earnubs committed:
Run prettier

earnubs committed:
Exclude querySelector test from envs with no window object

earnubs committed:
Run prettier

lmiller1990 committed:
Merge pull request #1372 from vuejs/docs/issue-1060-add-libraries-links

(docs): Add guide for some useful libraries for testing

kazupon committed:
fix: translation missing (#1374)

closes #1267

lmiller1990 committed:
(docs): update webpack link

lmiller1990 committed:
Merge pull request #1320 from earnubs/template-query-selector-support

Template querySelector support

hiendv committed:
test(wrapper): split disabled attribute tests for invalid and invalid elements

Commit:
Adding notes about async vs. sync mode. Adding placeholder for mocking transitions section

hiendv committed:
docs: add an example for slots components with props

lmiller1990 committed:
(docs): add notes on mocking transitions and run linter

lmiller1990 committed:
(docs): improve guide

Commit:
Update for PR comments

Commit:
Update for PR comments

Commit:
(docs): apply feedback and fix typos

JessicaSachs committed:
Merge pull request #1362 from hiendv/fix-disabled-attr

fix: whitelist tags to be ignored with `disabled` attribute

JessicaSachs committed:
Merge pull request #1375 from JessicaSachs/docs/beta30-notes

Notes about 1.0.0-beta.30

JessicaSachs committed:
Merge pull request #1373 from vuejs/docs/issue-1192-vue-test-utils-without-a-build-step

Guide on using vue test utils without a build step

hiendv committed:
docs: simplify options API for slots

hiendv committed:
chore: prettier docs/api

Commit:
Fix linting

Commit:
Revert flow to fix issues

JessicaSachs committed:
Merge pull request #1376 from hiendv/docs-api-options-slots

docs: add an example for slots components with props

JessicaSachs committed:
Merge pull request #1354 from Incognitus-Io/dev

Add provided/inject support for Composition API components

Commit:
fix: use the keyCodes provided in options from wrapper.trigger

Previously wrapper.trigger('keydown', { keyCode: 65 }) would trigger a keydown event with a keyCode
of 0. This commit fixes that bug.

PR closes vuejs/vue-test-utils#1285
PR closes vuejs/vue-test-utils#1295

JessicaSachs committed:
Merge pull request #1248 from winniehell/winniehell-wrapper-selector

feat: Store selector in Wrapper.find() / .findAll()

vvanpo committed:
test: ensure Wrapper.setChecked() is a no-op for checked checkboxes

vvanpo committed:
test: ensure Wrapper.setChecked() is a no-op for selected radios

Selecting already-selected radio buttons or checking already-checked
checkboxes should not trigger events.

vvanpo committed:
fix: ensure Wrapper.setChecked() idempotence

vvanpo committed:
test: fix Wrapper.setChecked() test compatibility

Older versions of Vue use the `click` DOM event for updating
radio/checkbox `v-model`s.

vvanpo committed:
test: ensure Wrapper.setSelected() idempotency

vvanpo committed:
fix: ensure Wrapper.setSelected() idempotency

vvanpo committed:
style: fix linting errors

vvanpo committed:
test: follow convention for scenario statements

Co-Authored-By: Adrià Fontcuberta <afontcu@gmail.com>

Commit:
Merge pull request #1378 from vuejs/fix/issue-1285-custom-keycodes-failing

fix: use the keyCodes provided in options from wrapper.trigger

afontcu committed:
Fix failing formatter

afontcu committed:
docs(mocha-webpack): replace mocha-webpack with mochapack

NataliaTepluhina committed:
docs: added example usage for localVue

JessicaSachs committed:
Merge pull request #1381 from NataliaTepluhina/docs/fix-localVue-example

Docs: added example usage for localVue

rahulcs committed:
docs: Add example for 'listeners' mount option

lmiller1990 committed:
(docs): include guide about testing with node

Commit:
chore: remove gh-pages

lmiller1990 committed:
Merge pull request #1388 from vuejs/chore/remove-gh-pages

chore: remove gh-pages

lmiller1990 committed:
Merge pull request #1380 from vvanpo/set-selected

fix: Ensure setChecked() and setSelected() only trigger DOM events when state is changed (fix #1339)

lmiller1990 committed:
Merge branch 'LinusBorg-docs-scopedSlots-1' of https://github.com/vuejs/vue-test-utils into LinusBorg-docs-scopedSlots-1

lmiller1990 committed:
(docs): run prettier

lmiller1990 commented:
I tried to rebase and encountered the same problem described above, but did not notice until pushing the commits. So I just made a fresh fork of dev and copied the content over. It's here: #1389.

Going to close this one and get the other one merged.
